modules/bucket-events/main.tf

```diff
@@ -0,0 +1,213 @@
+    "us" : "us-central1",
+    "eu" : "europe-west1",
+    "asia" : "asia-east1",
+  }, local.lowercase, local.lowercase)
+}
+
+resource "random_string" "service-suffix" {
+  length  = 4
+  upper   = false
+  special = false
+}
+
+// A dedicated service account for the trampoline service.
+resource "google_service_account" "service" {
+  project = var.project_id
+
+  account_id   = "${var.name}-${random_string.service-suffix.result}"
+  display_name = "Service account for ${var.bucket} trampoline service"
+}
+
+resource "random_string" "delivery-suffix" {
+  length  = 4
+  upper   = false
+  special = false
+}
+
+// A dedicated service account for this subscription.
+resource "google_service_account" "delivery" {
+  project = var.project_id
+
+  account_id   = "${var.name}-${random_string.delivery-suffix.result}"
+  display_name = "Delivery account for ${var.bucket} events"
+}
+
+// Lookup the identity of the pubsub service agent.
+resource "google_project_service_identity" "pubsub" {
+  provider = google-beta
+  project  = var.project_id
+  service  = "pubsub.googleapis.com"
+}
+
+// Authorize Pub/Sub to impersonate the delivery service account to authorize
+// deliveries using this service account.
+// NOTE: we use binding vs. member because we expect nothing but pubsub to be
+// able to assume this identity.
+resource "google_service_account_iam_binding" "allow-pubsub-to-mint-tokens" {
+  service_account_id = google_service_account.delivery.name
+
+  role    = "roles/iam.serviceAccountTokenCreator"
+  members = ["serviceAccount:${google_project_service_identity.pubsub.email}"]
+}
+
+module "audit-delivery-serviceaccount" {
+  source = "../audit-serviceaccount"
+
+  project_id      = var.project_id
+  service-account = google_service_account.delivery.email
+
+  # The absence of authorized identities here means that
+  # nothing is authorized to act as this service account.
+  # Note: Cloud Pub/Sub's usage doesn't show up in the
+  # audit logs.
+
+  notification_channels = var.notification_channels
+}
+
+module "this" {
+  source     = "../regional-go-service"
+  project_id = var.project_id
+  name       = var.name
+  regions = {
+    (local.region) : var.regions[local.region]
+  }
+
+  service_account = google_service_account.service.email
+  containers = {
+    "trampoline" = {
+      source = {
+        working_dir = path.module
+        importpath  = "./cmd/trampoline"
+      }
+      ports = [{ container_port = 8080 }]
+      env = [{
+        name  = "INGRESS_URI"
+        value = module.trampoline-emits-events.uri
+      }]
+    }
+  }
+
+  notification_channels = var.notification_channels
+}
+
+// Authorize the Pub/Sub topic to deliver events to the service.
+module "authorize-delivery" {
+  source = "../authorize-private-service"
+
+  project_id = var.project_id
+  region     = local.region
+  name       = var.name
+
+  service-account = google_service_account.delivery.email
+}
+
+resource "google_pubsub_topic" "dead-letter" {
```

Comment:
I think the `cloudevent-trigger` surfaces this as an output

Reply:
I don't get to use a trigger though, I'm dealing with a raw PubSub topic and a GCS notification. Let me know if I'm missing something though.

Reply:
I was just saying that it might make sense to output it in a similar way here